WorkloadManagementSystem/DB/TaskQueueDB.py

```diff
@@ -711,7 +711,15 @@ def __generateTQMatchSQL(self, tqMatchDict, numQueuesToGet=1, negativeCond=None)
       self.log.debug("Evaluating field %s" % field)
       # It has to be %ss , with an 's' at the end because the columns names
       # are plural and match options are singular
-      if tqMatchDict.get(field):
+
+      # Just treating the (not so) special case of no Tag, No RequiredTag, No BannedTag
+      if field in tagMatchFields \
+         and field not in tqMatchDict \
+         and 'Required' + field not in tqMatchDict \
+         and 'Banned' + field not in tqMatchDict:
+        tqMatchDict[field] = "''"
```

Comment:
This looks to work but the implementation is strange. You introduce a "fake" tag "''" which is not in starValues. This forces to add TagSQL condition which will always evaluate to False for TQs defining any tag. But if you just drop these lines and also drop "or not fv" in the line 737 below, then the result will be the same. 

The only wild card should be ANY/any explicitly defined by the site. No "''", '""', '' values should be used to avoid confusion.

Reply:
I pushed a change that removes "" or "''" from starValues